src/common/queries/useAccountBalance.ts

```diff
@@ -1,56 +1,45 @@
-import {
-  UseQueryOptions,
-  UseSuspenseQueryOptions,
-  useQuery,
-  useSuspenseQuery,
-} from '@tanstack/react-query';
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
 
-import { AddressBalanceResponse } from '@stacks/stacks-blockchain-api-types';
-
-import { useApi } from '../api/useApi';
+import { getErrorMessage } from '../../api/getErrorMessage';
+import { useApiClient } from '../../api/useApiClient';
 import { ONE_MINUTE } from './query-stale-time';
 
 const ACCOUNT_BALANCE_QUERY_KEY = 'accountBalance';
 
-export function useAccountBalance(
-  address?: string,
-  options: Omit<UseQueryOptions<any, any, AddressBalanceResponse, any>, 'queryKey' | 'queryFn'> = {}
-) {
-  const api = useApi();
+export function useAccountBalance(address?: string) {
+  const apiClient = useApiClient();
   return useQuery({
     queryKey: [ACCOUNT_BALANCE_QUERY_KEY, address],
-    queryFn: () => {
+    queryFn: async () => {
       if (!address) return undefined;
-      const response = api.accountsApi.getAccountBalance({
-        principal: address,
+      const { data, error } = await apiClient.GET('/extended/v1/address/{principal}/balances', {
+        params: { path: { principal: address } },
       });
-      return response;
+      if (error) {
+        throw new Error(getErrorMessage(error));
+      }
+      return data;
     },
     staleTime: ONE_MINUTE,
     enabled: !!address,
-    ...options,
   });
 }
 
-export function useSuspenseAccountBalance(
-  address?: string,
-  options: Omit<
-    UseSuspenseQueryOptions<any, any, AddressBalanceResponse, any>,
-    'queryKey' | 'queryFn'
-  > = {}
-) {
-  const api = useApi();
+export function useSuspenseAccountBalance(address?: string) {
+  const apiClient = useApiClient();
   if (!address) throw new Error('Address is required');
   return useSuspenseQuery({
     queryKey: [ACCOUNT_BALANCE_QUERY_KEY, address],
-    queryFn: () => {
-      const response = api.accountsApi.getAccountBalance({
-        principal: address,
+    queryFn: async () => {
+      const { data, error } = await apiClient.GET('/extended/v1/address/{principal}/balances', {
+        params: { path: { principal: address } },
       });
-
-      return response;
+      if (error) {
+        throw new Error(getErrorMessage(error));
+      }
+      return data;
     },
     staleTime: ONE_MINUTE,
-    ...options,
+    refetchOnWindowFocus: true,
```

Comment:
let's add a comment here why this is always set to true, such as we want the ui to always try to update the principal balance.